yarn-project/noir-contracts/src/libs/noir-aztec/src/oracle/notes.nr

```diff
@@ -83,11 +87,16 @@ fn get_notes<Note, N, S, NS>(
 ) -> [Note; S] {
     let fields = get_notes_oracle_wrapper(storage_slot, sort_by, sort_order, limit, offset, zero_fields);
     let num_notes = fields[0] as u32;
+    let contract_address = fields[1];
```

Comment:
I was wondering if it could be dangerous not to eventually constrain this to equal the current contract address but I think it shouldn't be problematic :thinking:

Reply:
That's a very good point! We should probably check the address and storage slot in the aztec library. And the app only needs to check the custom note fields.